pymongo/logger.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import os
+from typing import Any
+
+from bson import UuidRepresentation, json_util
+from bson.json_util import JSONOptions
+
+DEFAULT_DOCUMENT_LENGTH = 1000
+
+SENSITIVE_COMMANDS = [
+    "authenticate",
+    "saslStart",
+    "saslContinue",
+    "getnonce",
+    "createUser",
+    "updateUser",
+    "copydbgetnonce",
+    "copydbsaslstart",
+    "copydb",
+]
+HELLO_COMMANDS = ["hello", "ismaster", "isMaster"]
+
+REDACTED_FAILURE_FIELDS = ["code", "codeName", "errorLabels"]
+
+
+class StructuredMessage:
+    def __init__(self, **kwargs: Any):
+        document_length = int(os.getenv("MONGOB_LOG_MAX_DOCUMENT_LENGTH", DEFAULT_DOCUMENT_LENGTH))
+        if document_length < 0:
+            document_length = DEFAULT_DOCUMENT_LENGTH
```

Comment:
`test_configured_truncation_limit` caught this: since the max document length is configurable, and could be configured between different logging calls, we need to get `document_length` on every call of `LogMessage`.